core/amber/src/main/python/core/runnables/network_receiver.py

```diff
@@ -63,15 +63,18 @@ def data_handler(command: bytes, table: Table) -> int:
             :return: sender credits
             """
             data_header = PythonDataHeader().parse(command)
-            payload_type = data_header.payload_type
-            if payload_type == "data":
-                payload = DataFrame(table)
-            elif payload_type == "EndOfUpstream":
-                payload = MarkerFrame(EndOfUpstream())
+
+            payload_map = {
+                "Data": lambda table: DataFrame(table),
+                "State": lambda table: MarkerFrame(State(table)),
+                "StartOfUpstream": lambda _: MarkerFrame(StartOfUpstream()),
+                "EndOfUpstream": lambda _: MarkerFrame(EndOfUpstream()),
```

Comment:
it's hard to debug (trace the method invocation) with the map. use the matcher syntax?
```
match(item,
    TypeA, methodA,
    TypeB, methodB
)
```

Reply:
Fixed.